Funcky/Extensions/StringExtensions/SplitLines.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using Funcky.Monads;
+
+namespace Funcky.Extensions
+{
+    public static partial class StringExtensions
+    {
+        /// <summary>
+        /// Splits a string into individual lines lazily, by any new line (CR, LF, CRLF).
+        /// </summary>
+        /// <param name="text">The input text.</param>
+        /// <returns>A lazy IEnumerable containing the lines.</returns>
+        [Pure]
+        public static IEnumerable<string> SplitLines(this string text)
+            => text.SplitBy(ExtractBy(GetNextLine));
+
+        private static Option<SplitResult> GetNextLine(string text, int startIndex)
+            => GetNextLine()(text, startIndex);
+
+        private static ExtractElement GetNextLine()
+            => (text, startIndex)
+                => text
+                    .IndexOfAnyOrNone(new[] { '\r', '\n' }, startIndex)
```

Comment:
I'd extract the newline chars into a constant

Reply:
Addressed in a3319a50